src/python/pants/backend/go/util_rules/build_go_pkg.py

```diff
@@ -165,12 +123,21 @@ async def build_target(
         "-o",
         "__pkg__.a",
     ]
-    if symabis_digest:
-        compile_command.extend(["-symabis", "./gensymabis"])
-        input_digest = await Get(
-            Digest,
-            MergeDigests([input_digest, symabis_digest]),
+
+    input_digest = await Get(Digest, MergeDigests([gathered_imports.digest, source_files_digest]))
+
+    assembly_digests = None
+    if resolved_package.s_files:
+        assembly_setup = await Get(
+            AssemblyPreCompilation,
+            AssemblyPreCompilationRequest(
+                input_digest, resolved_package.s_files, source_files_subpath
+            ),
         )
+        input_digest = assembly_setup.merged_compilation_input_digest
+        assembly_digests = assembly_setup.assembly_digests
+        compile_command.extend(assembly_setup.EXTRA_COMPILATION_ARGS)
```

Comment:
Nothing to do now, but a note for a future refactor: If this is ever refactored later so that there is a `CompileGoFilesRequest` request type, that type can take a `symabis` digest as an attribute and handle appending the compiler flag (which keeps knowledge of `go tool compile` in one place versus the way it is split between this rule and the new assembly rules).

Tangentially, I am thinking we might want a `CompileGoFIlesRequest` later for test support because:
1. Regular `.go` files and test `.go` files can be compiled in separate batches via that request (and the resulting output combined into a single package archive for the test version of a package).
2. Building the test name inference program will need a way to compile Go files for use by the plugin. Although maybe existing binary support could leveraged for that. (Imagine embedding a Go Pants project in the Pants source and generating targets for that project that are only addressable by rules and not by the user.)